### PR TITLE
feat(curriculum): Add Basic JS Scrimba links

### PR DESCRIPTION
**Pull request by beaucarnes**
- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x] My changes do not use shortened URLs or affiliate links.

Closes #XXXXX

### CODE REVIEW

**Comment by raisedadead** [comment]
However, this is an incredible amount of help content. Thanks a lot to @beaucarnes for recording these.

/cc @freeCodeCamp/moderators for visibility. Just a heads up. This PR is going to touch a lot of the files in the curriculum.

**Comment by lynxlynxlynx** [comment]
Please don't merge until the hactoberfest curriculum prs are dealt with!

**Comment by QuincyLarson** [comment]
@beaucarnes Awesome! Once we get Master live, we'll merge and deploy this. Thanks again for all your awesome work on this! These videos are going to be a true windfall for all the people working through the curriculum.

**Comment by RandellDawson** [comment]
@raisedadead I wrote a script yesterday which validated all of the scrimba links produced a valid response (no 404s).  Do you want someone (or a group of us) to actually check each scrimba link and validate it matches up with the corresponding challenge or would a random sample of say 25-30 be good enough?

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/finding-a-remainder-in-javascript.english.md
Comment (beaucarnes) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/finding-a-remainder-in-javascript.english.md]: There is supposed to be a videoURL for this one. I will add it. Thanks for catching this.

**Comment by QuincyLarson** [comment]
@RandellDawson Yes - if you can just go through and click them all to make sure the match up, that would be a huge help. Last time a few of the links got mixed up and we didn't find out about it until after they'd been live for several weeks.

**Comment by RandellDawson** [comment]
@QuincyLarson I will work on this today.  If everything matches up, are you ok with me merging or should I just marked approved and tag you or @raisedadead?

**Comment by raisedadead** [comment]
> ... If everything matches up, are you ok with me merging or should I just marked approved and tag you or @raisedadead?

Go ahead and approve when you are done with the QA. I'll take one look just in case. Thanks a lot for all your help.

**Comment by RandellDawson** [comment]
@raisedadead What about the other 3 earlier PRs?  Are we ok merging this one ahead of those?  I don't believe there will be any merge conflicts for them.

**Comment by raisedadead** [comment]
> What about the other 3 earlier PRs?

- [ ] https://github.com/freeCodeCamp/freeCodeCamp/pull/27654 This one is conflicting with this PR? And you are awaiting user feedback? I do not see any request for changes there? Did you mean https://github.com/freeCodeCamp/freeCodeCamp/pull/27564? If yes, and if does conflict, I guess we will just have to resolve a conflict on that one later.
- [x] https://github.com/freeCodeCamp/freeCodeCamp/pull/27778 is not a blocker, its already conflicted and needs more discussion anyways.
- [ ] https://github.com/freeCodeCamp/freeCodeCamp/pull/34345 is waiting for feedback from us. And it indeed needs some through investigation, that the OP have noted. I think we can close and ask them to log an issue, that we can investigate. It's not just about fixing one test case or a challenge. There is a much deeper issue that we need to address there I think.

**Comment by raisedadead** [comment]
With that, I'll let you QA this PR and approve when you are happy with the changes. Thanks.

**Review by RandellDawson — CHANGES_REQUESTED** [review]
@raisedadead  - I only found one `videoUrl` which did not match for the given challenge (see above).  

I actually watched about 40 of the videos (first 10-15 seconds) and also noted above, a challenge where the challenge seed code is different than the video's code.  Finding this made me realize there is no formal process to notify Beau of any challenges which change significantly.  For example,  https://github.com/freeCodeCamp/freeCodeCamp/pull/32355 is going to significantly change the related challenge.  If there was already a videoUrl for that challenge, how would Beau know to create a new video?  Are we just going to wait until someone creates an issue for these situations?   

Using the Probot, we should be able to parse any merged challenge files and check for changes in the challenge description, instructions, or Challenge Seed sections.  If any are detected, then it could send Beau and email or automatically create an issue on GitHub which would be assigned to him.  But until then, I just wanted to make sure everyone is aware of what could happen.  Also, we could have the bot tag Beau on any PR which does make changes to any of those same sections, so he could be aware of the possible changes and give his opinion before any changes are finalized and merged.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/local-scope-and-functions.english.md
Comment (beaucarnes) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/local-scope-and-functions.english.md]: I occasionally removed the comments and 'use strict' lines when I created the videos since they didn't impact the code. I figured it was better to simplify the code when demonstrating so it would be easier to understand.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/add-two-numbers-with-javascript.english.md
Comment (RandellDawson) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/add-two-numbers-with-javascript.english.md]: @beaucarnes Is this other scrimba url I found the correct one to use?

**Comment by QuincyLarson** [comment]
@RandellDawson Awesome! Thanks for checking these!

Yes, as Beau mentioned, he will often modify the code a bit to simplify it.

I don't think Beau will need to record new video unless there is significant change to a challenge, though Probot task would help notify Beau so he can make the determination. We would also want to detect completely new challenges so Beau can be reminded to go in and create those Scrimbas as well - probably in batches once every few months.

@beaucarnes Once you've updated that Scrimba URL @RandellDawson found, it sounds like this is good to merge!

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/add-two-numbers-with-javascript.english.md
Comment (beaucarnes) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/add-two-numbers-with-javascript.english.md]: Yup. I just updated the code.

**Review by RandellDawson — APPROVED** [review]

**Comment by RandellDawson** [comment]
OK, the one link has been corrected, so I am merging this one.  

@beaucarnes Really fantastic work making all these videos!

**Pull request merged**
